frontend/packages/console-dynamic-plugin-sdk/src/app/modal-support/ModalProvider.tsx

```diff
@@ -18,25 +23,50 @@ export const ModalContext = React.createContext<ModalContextValue>({
   closeModal: () => {},
 });
 
+type ComponentMap = {
+  [key: string]: {
+    Component: ModalComponent<UnknownProps>;
+    props: { [key: string]: any };
+  };
+};
+
 export const ModalProvider: React.FC = ({ children }) => {
   const [isOpen, setOpen] = React.useState(false);
   const [Component, setComponent] = React.useState<ModalComponent>();
   const [componentProps, setComponentProps] = React.useState({});
+  const [componentsMap, setComponentsMap] = React.useState<ComponentMap>({});
 
   const launchModal = React.useCallback<LaunchModal>(
-    (component, compProps) => {
-      setComponent(() => component);
+    (component, compProps, id = undefined) => {
```

Comment:
Nit - `id` will be undefined anyway when not provided by the caller:
```suggestion
    (component, compProps, id) => {
```